### DISCUSSION TITLE
Sonoff RF R2 Power V1.3 (BASIC) boards fail to connect

### DISCUSSION THREAD

**Discussion by RandMental** [arendst/Tasmota#16216] [Q&A] [answered]
Good morning,

Having spent many hours, I need assistance please.

I have three  Sonoff RF R2 Power V1.3  boards,   2x  BASIC and 1x  RF (with the 433 MHz daughter board.)  The boards looks the same expect for the RF daughter board.
I use a Silicon Labs CP210x USB to UART Bridge with 3.6V that powers the Sonoff boards 
I use esptool.exe V4.2.1 on Win11. 

Powering on with the button pressed, the two BASIC boards never connect,  no matter what I do.  The RF board however connects first time and I could load Tasmota.bin with no issues.

The two BASIC boards give this error, implying that it could not set the port correctly?"

**PS C:\esptool> esptool.py -p COM15 flash_id
esptool.py v4.2.1
Serial port COM15
Traceback (most recent call last):
  File "C:\Python310\Scripts\esptool.py-script.py", line 33, in <module>
    sys.exit(load_entry_point('esptool==4.2.1', 'console_scripts', 'esptool.py')())
  File "C:\Python310\lib\site-packages\esptool\__init__.py", line 1004, in _main
    main()
  File "C:\Python310\lib\site-packages\esptool\__init__.py", line 646, in main
    esp = esp or get_default_connected_device(
  File "C:\Python310\lib\site-packages\esptool\__init__.py", line 889, in get_default_connected_device
    _esp = detect_chip(
  File "C:\Python310\lib\site-packages\esptool\cmds.py", line 86, in detect_chip
    detect_port = ESPLoader(port, baud, trace_enabled=trace_enabled)
  File "C:\Python310\lib\site-packages\esptool\loader.py", line 286, in __init__
    self._port.write_timeout = DEFAULT_SERIAL_WRITE_TIMEOUT
  File "C:\Python310\lib\site-packages\serial\serialutil.py", line 392, in write_timeout
    self._reconfigure_port()
  File "C:\Python310\lib\site-packages\serial\serialwin32.py", line 222, in _reconfigure_port
    raise SerialException(
serial.serialutil.SerialException: Cannot configure port, something went wrong. Original message: PermissionError(13, 'A device attached to the system is not functioning.', None, 31)**

_
![image](https://user-images.githubusercontent.com/6272455/184479696-9de5a2ef-2a2b-44a5-85e1-a1e9a123aa5f.png)

**Comment by sfromis** [comment, marked as answer]
**PermissionError** usually means that you need to grant yourself permission to use serial. Or run as admin.

**Reply by RandMental** [reply]
I have tried running in Admin (elevated terminal) and marked esptool.exe to run as administrator:   no luck
Tried 3 versions of the CP210x drivers - 2014/2018/2022:  no luck

I remembered I have an Adafruit MCP2221 interface somewhere - WORKED the first time with web install!  https://tasmota.github.io/install/

**LESSON:  Don't use a CP210x USB-Serial interface, even if it works sometime (I have two and both reacted the same.)**

**Reply by sfromis** [reply]
I have another reason for avoiding CP210x adapters, lots of them are woefully underpowered for reliable flashing, unless they have an extra voltage regulator on the board to facilitate acting as a power supply. Issues here can cause esptool failure, but I'd expect different messages.

**Comment by mqu** [comment]
hello ; I have a bunch of 3 Sonoff basic R2 - v1.3 - 2022.03.09 ; I have been able to flash Tasmota-light firmware using Chrome (Chromium) flasher (https://tasmota.github.io/install/).

please note : DO NOT inverse (cross) TX and RX pins for this version (Sonoff Basic R2 - 1.3).

after install, here is some information from status page:
```
 - MAC: 48:55:19:73:73:1E -
       hasdware: sonoff basic R2 - 1.3
	ESP Chip Id	7566110 (ESP8285N08)
	Flash Chip Id	0x14325E (DOUT)
	Flash Size	1024 KB
	Program Flash Size	1024 KB
	Program Size	493 KB
	Free Program Space	508 KB
	Free Memory	29.7 KB
```
